consensus/safety-rules/src/safety_rules.rs

```diff
@@ -67,9 +71,11 @@ impl SafetyRules {
         self.verify_epoch(proposed_block.epoch(), &safety_data)?;
 
         self.verify_qc(proposed_block.quorum_cert())?;
-        proposed_block
-            .validate_signature(&self.epoch_state()?.verifier)
-            .map_err(|error| Error::InvalidProposal(error.to_string()))?;
+        if !self.skip_sig_verify {
```

Comment:
Wondering if it would help carry a `VerifiedBlock` struct like VerifiedEvent.

Reply:
the safety rules design is for trust boundary, not for code level safety. so if it's running remotely, it needs to verify before signing. when it's local, it can help but it'll duplicate the whole interfaces.

Reply:
i was supposing we may never go back to that world with different trust boundaries on the same machine? or may be...

Reply:
agree, but that's a much bigger change to make. I can clean it up on a separate pr

Reply:
sounds good. whenever this PR is ready, flip it from draft and I can approve.